erpnext/crm/doctype/opportunity/opportunity.py

```diff
@@ -36,6 +36,20 @@ def validate(self):
 		if not self.with_items:
 			self.items = []
 
+		else:
+			self.calculate_totals()
+
+	def calculate_totals(self):
+		total = base_total = 0
+		for item in self.get('items'):
+			item.amount = item.rate * item.qty
```

Comment:
![image](https://user-images.githubusercontent.com/24353136/127468460-9e082040-18e3-4c4b-85dd-94d8909633e0.png)

Keep the qty 1 by default

```python
Traceback (most recent call last):
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/app.py", line 66, in application
    response = frappe.api.handle()
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/api.py", line 54, in handle
    return frappe.handler.handle()
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/handler.py", line 29, in handle
    data = execute_cmd(cmd)
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/handler.py", line 65, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/__init__.py", line 1174, in call
    return fn(*args, **newargs)
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/desk/form/save.py", line 20, in savedocs
    doc.save()
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 282, in save
    return self._save(*args, **kwargs)
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 304, in _save
    self.insert()
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 235, in insert
    self.run_before_save_methods()
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 961, in run_before_save_methods
    self.run_method("validate")
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 855, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 1148, in composer
    return composed(self, method, *args, **kwargs)
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 1131, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/Users/ruchamahabal/dev-bench/apps/frappe/frappe/model/document.py", line 849, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "/Users/ruchamahabal/dev-bench/apps/erpnext/erpnext/crm/doctype/opportunity/opportunity.py", line 40, in validate
    self.calculate_totals()
  File "/Users/ruchamahabal/dev-bench/apps/erpnext/erpnext/crm/doctype/opportunity/opportunity.py", line 45, in calculate_totals
    item.amount = item.rate * item.qty
TypeError: unsupported operand type(s) for *: 'int' and 'NoneType'
```